### PR TITLE
Fix HTTP/REST clients HTTP Content-Type header parsers.

### PR DESCRIPTION
**Pull request by cheatfate**

### CODE REVIEW

File: beacon_chain/spec/eth2_apis/eth2_rest_serialization.nim [hunk 1]
```diff
@@ -2392,10 +2402,20 @@ proc decodeBytes*[T: DecodeTypes](t: typedesc[T], value: openArray[byte],
...
+proc decodeBytes*[T: SszDecodeTypes](
+       t: typedesc[T],
+       value: openArray[byte],
+       contentType: Opt[ContentTypeData],
+       updateRoot = true
+     ): RestResult[T] =
+
+  let mediaType =
+    if contentType.isNone():
+      OctetStreamMediaType
```
Comment (zah) [inline beacon_chain/spec/eth2_apis/eth2_rest_serialization.nim]: Are there any places in the code where this function can be called with a `None` value? I can't think of any API end-point where the content type is assumed to be SSZ by default, so it seems to me that this function should not even accept a `contentType` parameter and all its usage sites should be guarded to ensure that the SSZ content type was specified before calling the function.

In particular, the default choice to assume `OctetStreamMediaType` if no content type is specified is something that looks like a API spec violation.

**Comment by etan-status** [comment]
Hmm, while the Content-Type with charset now parses fine, there's now a weird REST JSON parsing issue, could it be related to the fix, or separate?

```
% build/nimbus_beacon_node trustedNodeSync --network:sepolia --backfill=false --data-dir="$HOME/Downloads/nimbus/data/sepolia" "--trusted-node-url=https://lodestar-sepolia.chainsafe.io" --blockId=234368 --log-level=DEBUG
NOT 2022-09-17 14:29:45.298+02:00 Starting trusted node sync                 databaseDir=/Users/etan/Downloads/nimbus/data/sepolia/db restUrl=https://lodestar-sepolia.chainsafe.io blockId=234368 backfill=false reindex=false
INF 2022-09-17 14:29:45.486+02:00 Writing genesis state                      stateRoot=fb9afe32 genesis_validators_root=d8ea171f
INF 2022-09-17 14:29:45.522+02:00 Writing genesis block                      blockRoot=fb9b64fe blck="(slot: 0, proposer_index: 0, parent_root: \"00000000\", state_root: \"fb9afe32\", eth1data: (deposit_root: 0000000000000000000000000000000000000000000000000000000000000000, deposit_count: 0, block_hash: 0000000000000000000000000000000000000000000000000000000000000000), graffiti: \"\", proposer_slashings_len: 0, attester_slashings_len: 0, attestations_len: 0, deposits_len: 0, voluntary_exits_len: 0, sync_committee_participants: -1)"
DBG 2022-09-17 14:29:45.525+02:00 Sending REST request to remote server      remote=lodestar-sepolia.chainsafe.io:443 request=/eth/v1/beacon/blocks/genesis/root http_method=GET
DBG 2022-09-17 14:29:46.097+02:00 Got REST response headers from remote server status=200 http_method=GET remote=lodestar-sepolia.chainsafe.io:443 request=/eth/v1/beacon/blocks/genesis/root connection=1
DBG 2022-09-17 14:29:46.097+02:00 Received REST response body from remote server status=200 http_method=GET remote=lodestar-sepolia.chainsafe.io:443 request=/eth/v1/beacon/blocks/genesis/root connection=1 contentType="application/json; charset=\"utf-8\"" size=77
DBG 2022-09-17 14:29:46.098+02:00 Failed to deserialize REST JSON data       err="<data>(1, 9) Unexpected token 'tkString' in place of 'object start bracket'" data="{\"data\":\"0xfb9b64fe445f76696407e1e3cc390371edff147bf712db86db6197d4b31ede43\"}"
ERR 2022-09-17 14:29:46.098+02:00 Unable to download genesis block root      error="Unable to decode REST response, error [Serialization error]" restUrl=https://lodestar-sepolia.chainsafe.io
```

**Comment by cheatfate** [comment]
> Hmm, while the Content-Type with charset now parses fine, there's now a weird REST JSON parsing issue, could it be related to the fix, or separate?
> 
> ```
> % build/nimbus_beacon_node trustedNodeSync --network:sepolia --backfill=false --data-dir="$HOME/Downloads/nimbus/data/sepolia" "--trusted-node-url=https://lodestar-sepolia.chainsafe.io" --blockId=234368 --log-level=DEBUG
> NOT 2022-09-17 14:29:45.298+02:00 Starting trusted node sync                 databaseDir=/Users/etan/Downloads/nimbus/data/sepolia/db restUrl=https://lodestar-sepolia.chainsafe.io blockId=234368 backfill=false reindex=false
> INF 2022-09-17 14:29:45.486+02:00 Writing genesis state                      stateRoot=fb9afe32 genesis_validators_root=d8ea171f
> INF 2022-09-17 14:29:45.522+02:00 Writing genesis block                      blockRoot=fb9b64fe blck="(slot: 0, proposer_index: 0, parent_root: \"00000000\", state_root: \"fb9afe32\", eth1data: (deposit_root: 0000000000000000000000000000000000000000000000000000000000000000, deposit_count: 0, block_hash: 0000000000000000000000000000000000000000000000000000000000000000), graffiti: \"\", proposer_slashings_len: 0, attester_slashings_len: 0, attestations_len: 0, deposits_len: 0, voluntary_exits_len: 0, sync_committee_participants: -1)"
> DBG 2022-09-17 14:29:45.525+02:00 Sending REST request to remote server      remote=lodestar-sepolia.chainsafe.io:443 request=/eth/v1/beacon/blocks/genesis/root http_method=GET
> DBG 2022-09-17 14:29:46.097+02:00 Got REST response headers from remote server status=200 http_method=GET remote=lodestar-sepolia.chainsafe.io:443 request=/eth/v1/beacon/blocks/genesis/root connection=1
> DBG 2022-09-17 14:29:46.097+02:00 Received REST response body from remote server status=200 http_method=GET remote=lodestar-sepolia.chainsafe.io:443 request=/eth/v1/beacon/blocks/genesis/root connection=1 contentType="application/json; charset=\"utf-8\"" size=77
> DBG 2022-09-17 14:29:46.098+02:00 Failed to deserialize REST JSON data       err="<data>(1, 9) Unexpected token 'tkString' in place of 'object start bracket'" data="{\"data\":\"0xfb9b64fe445f76696407e1e3cc390371edff147bf712db86db6197d4b31ede43\"}"
> ERR 2022-09-17 14:29:46.098+02:00 Unable to download genesis block root      error="Unable to decode REST response, error [Serialization error]" restUrl=https://lodestar-sepolia.chainsafe.io
> ```

So i have performed same request using `curl`.
```
curl https://lodestar-sepolia.chainsafe.io/eth/v1/beacon/blocks/genesis/root
{"data":"0xfb9b64fe445f76696407e1e3cc390371edff147bf712db86db6197d4b31ede43"}
```
As you can see response generated by `lodestar` do not follows specification https://ethereum.github.io/beacon-APIs/#/Beacon/getBlockRoot

Because minimal response should be
```
{"data": {"root": "0xfb9b64fe445f76696407e1e3cc390371edff147bf712db86db6197d4b31ede43"}}
```

**Comment by etan-status** [comment]
Oh, that indeed explains the error. 

@wemeetagain could you please update Lodestar with correct response format for `getBlockRoot` endpoint?

**Review by etan-status — APPROVED** [review]

**Comment by dapplion** [comment]
- Lodestar was not complying with getBlockRoot response type until this PR https://github.com/ChainSafe/lodestar/pull/4493

Not included in LTS, but is in chainsafe/lodestar:next

**Pull request merged**